src/test/java/jenkins/plugins/openstack/PluginTestRule.java

```diff
@@ -114,10 +116,31 @@ public static SlaveOptions dummySlaveOptions() {
         }
         return new SlaveOptions(
                 new BootSource.VolumeSnapshot("id"), "hw", "nw1,mw2", "dummyUserDataId", 1, 2, "pool", "sg", "az", 1, null, 10,
-                "jvmo", "fsRoot", LauncherFactory.JNLP.JNLP, 1
+                "jvmo", "fsRoot", LauncherFactory.JNLP.JNLP, mkListOfNodeProperties(1, 2), 1
         );
     }
 
+    public static List<NodeProperty<Node>> mkListOfNodeProperties(int... npTypes) {
+        final Builder<NodeProperty<Node>> b = ImmutableList.builder();
+        for (int number : npTypes) {
+            b.add(mkNodeProperty(number));
+        }
+        return b.build();
+    }
+
+    public static NodeProperty<Node> mkNodeProperty(int number) {
+        switch (number) {
+        case 1:
+            return new NodePropertyOne();
+        case 2:
+            return new NodePropertyOne();
```

Comment:
This should probably return different implementations.

Reply:
Oops.  Yup - good spot...

Reply:
Fixed in https://github.com/jenkinsci/openstack-cloud-plugin/pull/270/commits/e0ebacc677cb89350fe1ee461973ef10cb3a1495